DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -154,28 +174,33 @@ public bool NewEmailAddressIsValid(string emailAddress, int? adminUserId, int? d
             return adminUsersWithNewEmail == null && delegateUsersWithNewEmail.Count(u => u.CentreId == centreId) == 0;
         }
 
-        private static bool UserEmailHasChanged(User? adminUser, string emailAddress)
+        private static bool UserEmailHasChanged(User? user, string emailAddress)
         {
-            return adminUser != null && adminUser.EmailAddress != emailAddress;
+            return user != null && user.EmailAddress != emailAddress;
         }
 
-        private (AdminUser?, List<DelegateUser>) GetVerifiedLinkedUsersAccounts
-        (
+        public (AdminUser?, List<DelegateUser>) GetVerifiedLinkedUsersAccounts(
             int? adminId,
             int? delegateId,
             string password
         )
         {
             var (loggedInAdminUser, loggedInDelegateUser) = GetUsersById(adminId, delegateId);
 
-            var signedInEmail = loggedInAdminUser?.EmailAddress ?? loggedInDelegateUser?.EmailAddress;
+            var signedInEmailIfAny = loggedInAdminUser?.EmailAddress ?? loggedInDelegateUser?.EmailAddress;
 
-            if (signedInEmail == null)
+            if (string.IsNullOrWhiteSpace(signedInEmailIfAny))
             {
-                return (null, new List<DelegateUser>());
+                return loginService.VerifyUsers(
+                    password,
+                    loggedInAdminUser,
+                    loggedInDelegateUser != null
+                        ? new List<DelegateUser> { loggedInDelegateUser }
+                        : new List<DelegateUser>()

```

Comment:
Super minor, but could we pull this into loggedInDelegateUsers var for better readability?